omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,236 @@
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("uuid");
+		description.addProperty("identifierType");
+		description.addProperty("location");
+		description.addProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("location");
+		description.addProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public SimpleObject create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		String locationUuid = postBody.get("location").toString();
+		String manualEntryEnabled = postBody.get("manualEntryEnabled").toString();
+		String automaticGenerationEnabled = postBody.get("automaticGenerationEnabled").toString();
+		String sourceUuid = postBody.get("source").toString();
+		String identifierUuid = postBody.get("identifierType").toString();
+		AutoGenerationOption autoGenerationOption = new AutoGenerationOption();
+		
+		if ((locationUuid != null && manualEntryEnabled != null && automaticGenerationEnabled != null && sourceUuid != null
+		        && identifierUuid != null)) {
+			IdentifierSource identifierSource = identifierSourceService.getIdentifierSourceByUuid(sourceUuid);
+			Location location = Context.getLocationService().getLocationByUuid(locationUuid);
+			PatientIdentifierType patientIdentifierType = Context.getPatientService()
+			        .getPatientIdentifierTypeByUuid(identifierUuid);
+			autoGenerationOption.setLocation(location);
+			autoGenerationOption.setSource(identifierSource);
+			autoGenerationOption.setIdentifierType(patientIdentifierType);
+			autoGenerationOption.setManualEntryEnabled(Boolean.parseBoolean(manualEntryEnabled));
+			autoGenerationOption.setAutomaticGenerationEnabled(Boolean.parseBoolean(automaticGenerationEnabled));
+			
+		} else if (locationUuid != null && manualEntryEnabled != null && automaticGenerationEnabled != null
+		        && sourceUuid == null && identifierUuid != null) {
+			Location location = Context.getLocationService().getLocationByUuid(locationUuid);
+			PatientIdentifierType patientIdentifierType = Context.getPatientService()
+			        .getPatientIdentifierTypeByUuid(identifierUuid);
+			autoGenerationOption.setLocation(location);
+			autoGenerationOption.setIdentifierType(patientIdentifierType);
+			autoGenerationOption.setManualEntryEnabled(Boolean.parseBoolean(manualEntryEnabled));
+			autoGenerationOption.setAutomaticGenerationEnabled(Boolean.parseBoolean(automaticGenerationEnabled));
+			
+		} else {
+			throw new ResourceDoesNotSupportOperationException(
```

Comment:
The error thrown is a validation exception if any of the parameters is missing or invalid

Reply:
Validation exception will require a try catch block, but I can use the ResourceDoesNotSupportOperationException which comes with all the tooling to give a response, plus  updating an autoGenerationOption with missing parameters is not supported, at least one parameter must be present for an update operation